src/ipahealthcheck/ipa/certs.py

```diff
@@ -1021,7 +1021,16 @@ def check_agent(plugin, base_dn, agent_type):
             return
         ra_desc = raw_desc[0]
         ra_certs = entry.get('usercertificate')
-        if ra_desc != description:
+        (_version, exp_serial, exp_issuer, exp_subject) = \
+            ra_desc.split(';')
+        matched = all(
+            [
+                str(serial_number) == exp_serial,
+                DN(issuer) == DN(exp_issuer),
+                DN(subject) == DN(exp_subject),
```

Comment:
According to lines 976 and 977, issuer and subject are already of DN type so there is no need to use DN(issuer) and DN(subject) (but I agree it doesn't do any harm).

Reply:
I was being a bit paranoid here. If a value is already an RDN then the DN class simply copies it as-is so it doesn't create a lot of overhead.